src/app/dataset-view/dataset.component.ts

```diff
@@ -63,9 +77,34 @@ export class DatasetComponent extends BaseDatasetDataComponent implements OnInit
             _.isNil(this.datasetBasics) ||
             this.datasetBasics.name !== urlDatasetInfo.datasetName ||
             this.datasetBasics.owner.accountName !== urlDatasetInfo.accountName ||
-            this.datasetViewType === DatasetViewTypeEnum.Flows
+            [DatasetViewTypeEnum.Overview, DatasetViewTypeEnum.Metadata].includes(this.datasetViewType)
         ) {
-            this.requestMainData(urlDatasetInfo);
+            this.datasetService
+                .requestDatasetHeadBlockHash(urlDatasetInfo.accountName, urlDatasetInfo.datasetName)
+                .pipe(
+                    map((hashBlock: string) => {
+                        if (hashBlock !== this.currentHeadBlockHash && this.datasetBasics) {
+                            this.updateCache(this.datasetBasics);
+                        }
+                    }),
+                    switchMap(() => this.datasetService.requestDatasetMainData(urlDatasetInfo)),
+                    tap(() => {
+                        this.mainDatasetQueryComplete$.next(urlDatasetInfo);
+                    }),
+                    takeUntilDestroyed(this.destroyRef),
+                )
+                .subscribe();
+        }
+    }
```

Comment:
I can see that everything is re-queried after returning from History tab.
1) Open dataset Overview initially
2) Switch to History
3) Switch to Overview back - I can see main data is re-queried without any reason.

Reply:
I can't reproduce it, please check again

Reply:
Can't reproduce it now too, weird.